packages/flutter/lib/src/material/data_table.dart

```diff
@@ -0,0 +1,757 @@
+  bool get _debugInteractive => onTap != null;
+}
+
+/// A material design data table.
+///
+/// Displaying data in a table is expensive, because to lay out the
+/// table all the data must be measured twice, once to negotiate the
+/// dimensions to use for each column, and once to find the rendering
+/// given the results of the negotiation.
+///
+// /// For this reason, if you have a lot of data (say, more than a dozen
+// /// rows with a dozen columns, though the precise limits depend on the
+// /// target device), it is suggested that you use a [DataCard] which
+// /// automatically splits the data into multiple pages.
+// ///
+/// See also:
+///
+///  * [DataColumn]
+///  * [DataRow]
+///  * [DataCell]
+///  * <https://www.google.com/design/spec/components/data-tables.html>
+class DataTable extends StatelessWidget {
+  /// Creates a widget describing a data table.
+  ///
+  /// The [columns] argument must be a list of as many [DataColumn]
+  /// objects as the table is to have columns, ignoring the leading
+  /// checkbox column if any. The table must have at least one column.
+  /// 
+  /// The [rows] argument must be a list of as many [DataRow] objects
+  /// as the table is to have rows, ignoring the leading heading row.
+  /// There may be zero rows, but the rows argument must not be null.
+  ///
+  /// Each [DataRow] object in [rows] must have as many [DataCell]
+  /// objects in the [DataRow.cells] list as the table has columns.
+  ///
+  /// If the table is sorted, the column that provides the current
+  /// primary key should be specified by index in [sortColumn], 0
+  /// meaning the first column in [columns], 1 being the next one, and
+  /// so forth.
+  ///
+  /// The actual sort order can be specified using [sortAscending]; if
+  /// the sort order is ascending, this should be true (the default),
+  /// otherwise it should be false.
+  DataTable({
+    Key key,
+    List<DataColumn> columns,
+    this.sortColumn,
+    this.sortAscending: true,
+    this.rows
+  }) : columns = columns,
+       _onlyTextColumn = _initOnlyTextColumn(columns), super(key: key) {
+    assert(columns != null);
+    assert(columns.length > 0);
+    assert(sortAscending != null);
+    assert(rows != null);
+    assert(!rows.any((DataRow row) => row.cells.length != columns.length));
+    assert(sortColumn == null || (sortColumn >= 0 && sortColumn < columns.length));
+  }
+
+  /// The configuration and labels for the columns in the table.
+  final List<DataColumn> columns;
+
+  /// The current primary sort key's column.
+  ///
+  /// If non-null, indicates that the indicated column is the column
+  /// by which the data is sorted. The number must correspond to the
+  /// index of the relevant column in [columns].
+  ///
+  /// Setting this will cause the relevant column to have a sort
+  /// indicator displayed.
+  ///
+  /// When this is null, it implies that the table's sort order does
+  /// not correspond to any of the columns.
+  final int sortColumn;
```

Comment:
It's not a DataColumn because a single DataColumn can be used multiple times in a single DataTable.

sortColumnIndex makes sense though.